src/nvim/mark.c

```diff
@@ -645,31 +645,28 @@ fmark_T *getnextmark(pos_T *startpos, int dir, int begin_line)
 // until the mark is used to avoid a long startup delay.
 static void fname2fnum(xfmark_T *fm)
 {
-  char_u *p;
-
   if (fm->fname != NULL) {
     // First expand "~/" in the file name to the home directory.
     // Don't expand the whole name, it may contain other '~' chars.
-    if (fm->fname[0] == '~' && (fm->fname[1] == '/'
 #ifdef BACKSLASH_IN_FILENAME
-                                || fm->fname[1] == '\\'
+    if (fm->fname[0] == '~' && (fm->fname[1] == '/' || fm->fname[1] == '\\')) {
+#else
+    if (fm->fname[0] == '~' && (fm->fname[1] == '/')) {
 #endif
```

Comment:
I have though about this several times: Is it really worthy to create duplicate code to fix lint failures? Or is it better to add `// NOLINT` instead?

Reply:
Even though there is a bit of duplicate code, I find it so much easier to read.

The trailing `)) {` is very ugly.